### PR TITLE
Add databasing

### PR DESCRIPTION
**Pull request by IAmTomahawkx**
This adds database support, which allows roles to be controlled from the db, and also allows statistics tracking

### CODE REVIEW

**Comment by pikaninja** [comment]
appease the sarc

**Review by LyricLy — APPROVED** [review]

File: extensions/stats.py [hunk 1]
```diff
@@ -82,6 +86,74 @@ def condition(
         )
         await ctx.reply(fmt, mention_author=False)
 
+    @commands.command("serverstats")
```
Comment (pikaninja) [inline extensions/stats.py]: guildstats alias when

File: mafia/roles.py [hunk 2]
```diff
@@ -69,7 +71,7 @@ def __le__(self, other: AttackType):
 
 class Role(abc.ABC):
     # The ID that will be used to identify roles for config
-    id: int = -1
+    id: int = None
```
Comment (Phxntxm) [inline mafia/roles.py]:
```suggestion
    id: typing.Optional[int] = None
```

File: mafia/roles.py [hunk 3]
```diff
@@ -119,7 +121,7 @@ def __str__(self) -> str:
 
 
 class Citizen(Role):
-    id = 0
+    id = None
```
Comment (Phxntxm) [inline mafia/roles.py:124]: Since all of these for every role is the same as the Role class, they aren't really needed

File: mafia/roles.py [hunk 4]
```diff
@@ -527,3 +530,39 @@ def win_condition(self, game: MafiaGame, player: Player) -> bool:
...
+                    x.id = await conn.fetchval(
+                        "INSERT INTO roles (name, alignment, attack_level, defence_level) VALUES ($1, $2, $3, $4) RETURNING id",
+                        x.__name__,
+                        1 if x.is_citizen else (2 if x.is_independent else 3),
+                        x.attack_type and x.attack_type.value,
+                        x.defense_type and x.defense_type.value,
+                    )
+
+        return
+        raise RuntimeError(
```
Comment (Phxntxm) [inline mafia/roles.py:554]: Since there's a return above, this raise doesn't do anything

**Review by Phxntxm — CHANGES_REQUESTED** [review]

File: mafia/roles.py (hunk 3 above)
Comment (IAmTomahawkx) [reply mafia/roles.py:124]: this resolves https://canary.discord.com/channels/840698427755069470/840698427755069475/847639392469778463

Comment (IAmTomahawkx) [reply mafia/roles.py:124]: wontfix

**Review by Phxntxm — COMMENTED** [review]
Lgtm!

**Review by Phxntxm — COMMENTED** [review]
Lgtm!

**Review by XuaTheGrate — COMMENTED** [review]
Lgtm!

**Review by LyricLy — COMMENTED** [review]
Lgtm!

**Comment by pikaninja** [comment]
Looks weird to me!

**Comment by lmwk** [comment]
Looks like gibberish to me!

**Pull request merged**